shard.go

```diff
@@ -72,13 +77,45 @@ func (s *shard[T]) forceEvict() {
 	entriesEvicted := 0
 	for key, e := range s.entries {
 		if e.expiresAt.Before(cutoff) {
-			delete(s.entries, key)
+			s.unsafeDelete(key)
 			entriesEvicted++
 		}
 	}
 	s.reportEntriesEvicted(entriesEvicted)
 }
 
+// should be called with a lock.
+func (s *shard[T]) unsafeDelete(key string) {
```

Comment:
I like the `unsafe` naming convention for functions that should be called inside a lock context.